shell/browser/net/asar/asar_url_loader.cc

```diff
@@ -152,25 +152,25 @@ class AsarURLLoader : public network::mojom::URLLoader {
     // requests at the same time.
     base::File file(info.unpacked ? real_path : archive->path(),
                     base::File::FLAG_OPEN | base::File::FLAG_READ);
-    // The validator also needs a file handle so it can independently read
-    // additional bytes to validate hashes
-    base::File validator_file(info.unpacked ? real_path : archive->path(),
-                              base::File::FLAG_OPEN | base::File::FLAG_READ);
     auto file_data_source =
-        std::make_unique<mojo::FileDataSource>(std::move(file));
-    auto asar_validator = std::make_unique<AsarFileValidator>(
-        info.integrity, std::move(validator_file));
+        std::make_unique<mojo::FileDataSource>(file.Duplicate());
+    std::unique_ptr<mojo::DataPipeProducer::DataSource> readable_data_source;
     mojo::FileDataSource* file_data_source_raw = file_data_source.get();
-    AsarFileValidator* file_validator_raw = asar_validator.get();
-    auto filtered_data_source = std::make_unique<mojo::FilteredDataSource>(
-        std::move(file_data_source), std::move(asar_validator));
+    AsarFileValidator* file_validator_raw = nullptr;
+    if (info.integrity.has_value()) {
+      auto asar_validator = std::make_unique<AsarFileValidator>(
+          std::move(info.integrity.value()), std::move(file));
+      file_validator_raw = asar_validator.get();
+      readable_data_source.reset(new mojo::FilteredDataSource(
+          std::move(file_data_source), std::move(asar_validator)));
+    } else {
+      readable_data_source.reset(file_data_source.release());
```

Comment:
nit: i think this works & is a little more canonical?
```suggestion
      readable_data_source = std::move(file_data_source);
```